instagram/bind.py

```diff
@@ -1,14 +1,17 @@
-import urllib
-from oauth2 import OAuth2Request
+from .oauth2 import OAuth2Request
 import re
-from json_import import simplejson
+from .json_import import simplejson
+
+import six
+from six.moves.urllib.parse import quote
+
 
 re_path_template = re.compile('{\w+}')
 
 
 def encode_string(value):
     return value.encode('utf-8') \
-                        if isinstance(value, unicode) else str(value)
+        if isinstance(value, six.text_type) else str(value)
```

Comment:
`str(value)` will give you: 1. unicode type 2. representation of the original string:

```python
>>> str(b'pr\xc3\xb3ba')
"b'pr\\xc3\\xb3ba'"
```

Reply:
@homm Did you mean that `str` call is redundant?
If so - this code should work on py2 and py3, so your example for py2 will have different result

```
Python 2.7.10 (default, Jul 14 2015, 19:46:27)
[GCC 4.8.2] on linux
> str(b'pr\xc3\xb3ba')
'pr\xc3\xb3ba'
```

Reply:
> Python 2.7.10

???

Reply:
@homm Clarify question, please.